src/Interpreters/GlobalSubqueriesVisitor.h

```diff
@@ -205,10 +205,20 @@ class GlobalSubqueriesMatcher
     }
 
 private:
+    static bool should_be_executed_globally(const Data & data)
+    {
+        const Settings & settings = data.getContext()->getSettingsRef();
+        /// For parallel replicas we reinterpret JOIN as GLOBAL JOIN as a way to broadcast data
+        const bool enable_parallel_processing_of_joins
```

Comment:
I think we should also check whether `use_hedged_requests` is enabled to be consistent with `getParallelReplicasMode()` method inside Context.

Reply:
It seems it could even be replaced by a call to `canUseParallelReplicasOnInitiator()` and would keep things clean